src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,64 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_SIZE = 10;
+    private static final int INDEX_OF_NOTHING = -1;
+    private final Object[] keys;
+    private final Object[] values;
+    private int size;
+
+    public StorageImpl() {
+        keys = new Object[MAX_SIZE];
+        values = new Object[MAX_SIZE];
+        size = 0;
+    }
+
     @Override
     public void put(K key, V value) {
+        if (get(key) != null) {
+            updatePair(key, value);
+        } else {
+            addPair(key, value);
+        }
     }
 
+    @SuppressWarnings("unchecked")
     @Override
     public V get(K key) {
-        return null;
+        int index = getKeyIndex(key);
+        return (index != INDEX_OF_NOTHING)
+                ? (V) values[index] : null;
     }
 
     @Override
     public int size() {
-        return -1;
+        return size;
+    }
+
+    private int getKeyIndex(K key) {
+        for (int i = 0; i < keys.length; i++) {
+            if ((key == null && keys[i] == null)
+                    || (keys[i] != null && keys[i].equals(key))) {
```

Comment:
Let's extract it to a separate method.
```suggestion
            if (keysAreEqual(...)) {
```